src/DevTools/createMockNetworkLayer/index.ts

```diff
@@ -195,19 +197,24 @@ const inferUnionOrInterfaceType = (
   const possibleTypes = unionMemberTypes.map(type => type.name).join(", ")
   throw error(
     info,
-    ({ path }) =>
-      `Ambiguous object at path '${path}'. Add a __typename from this list: [${possibleTypes}]`
+    ({ path, operationName }) =>
+      `Ambiguous object at path '${path}' for operation '${operationName}'. Add a __typename from this list: [${possibleTypes}]`
   )
 }
 
 function error(
   info: GraphQLResolveInfo,
-  renderMessage: (args: { type: string; path: string }) => string
+  renderMessage: (args: {
+    type: string
+    path: string
+    operationName: string
+  }) => string
 ) {
   return new Error(
     renderMessage({
       path: responsePathAsArray(info.path).join("/"),
       type: info.returnType.inspect(),
+      operationName: get(info, i => i.operation.name.value, "(unknown)"),
```

Comment:
I don't have a test covering the "unknown" case. I don't really know what it would mean to not know the operation name. `info.operation.name` _is_ optional, though, so I took the safe route and used `get`.

Reply:
I guess this is for when you don’t name an operation (which we shouldn’t be doing). I.e.

```graphql 
query {
  # ...
}
```

Reply:
So perhaps it would be good to take that into account when printing the helpful messages.

Reply:
@alloy I wonder if that should be a lint rule instead?

Reply:
ah it already is